File: 239/main.py
# O(n^2) solution

class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        if nums is None or n <= 0:
            return None
        result = []
        
        for i in range(n - k + 1):
            temp = -10000
            for j in range(i, i + k):
                if nums[j] > temp:
                    temp = nums[j]
                    
            result.append(temp)
                
        return result

File: 239/main.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* maxSlidingWindow(int* nums, int numsSize, int k, int* returnSize){
    int n = numsSize;
    if(nums == NULL || n <= 0) {
        *returnSize = 0;
        return NULL;
    }
    *returnSize = n - k + 1;
    int *result = (int *)malloc((*returnSize) * sizeof(int));
    
    
    for(int i = 0; i < *returnSize; i++) {
        int temp = INT_MIN;
        for(int j = i; j < i + k; j++) {
            if(nums[j] > temp) {
                temp = nums[j];
            }
        }
        result[i] = temp;
    }
    
    return result;
}
